### PR TITLE
Updated deprecated ast.NameConstant with ast.Constant #6305

### LINKED ISSUE #6305: ast.NameConstant is deprecated and will be removed in Python 3.14; use ast.Constant instead
```
scrapy/utils/misc.py:249: DeprecationWarning: ast.NameConstant is deprecated and will be removed in Python 3.14; use ast.Constant instead
    value is None or isinstance(value, ast.NameConstant) and value.value is None
```

Also, if I understand it correctly `isinstance(value, ast.NameConstant)` is always False, as `ast.NameConstant()` returns an `ast.Constant` instance, so I wonder if this case is covered by a test.

### PR DESCRIPTION
**Pull request by ThunderMind2019**
Fixes #6305 

Updated deprecated ast.NameConstant with ast.Constant

### CODE REVIEW

**Comment by wRAR** [comment]
I assume this doesn't need new tests because it's covered by existing ones?

**Comment by marslanabdulrauf** [comment]
yes @wRAR

**Comment by ThunderMind2019** [comment]
yes @wRAR 
filePath: `scrapy/tests/test_utils_misc/test_return_with_argument_inside_generator.py`

File: scrapy/pipelines/media.py [hunk 1]
```diff
@@ -211,7 +211,7 @@ def _cache_result_and_execute_waiters(
             # minimize cached information for failure
             result.cleanFailure()
             result.frames = []
-            result.stack = None
+            result.stack = []
```
Comment (wRAR) [inline scrapy/pipelines/media.py:214]: Can you please revert this change? It's unrelated and has its own PR.

Comment (ThunderMind2019) [reply scrapy/pipelines/media.py]: done

**Review by wRAR — APPROVED** [review]
